src/Winton.DomainModelling.DocumentDb/ValueObjectDocument.cs

```diff
@@ -6,27 +6,29 @@
 
 namespace Winton.DomainModelling.DocumentDb
 {
-    internal sealed class ValueObjectDocument<TValueObject>
+    internal sealed class ValueObjectDocument<TValueObject, TDto>
         where TValueObject : IEquatable<TValueObject>
     {
+        public ValueObjectDocument(TValueObject valueObject, TDto dto)
```

Comment:
I think this should be a `static Create` method as it's only intended to be used for the first creation of one of these objects when you do not know the Id. Having it as a constructor might cause us to accidentally use this constructor in the future for recreating instances of this when the Id should be set. Obviously this class is internal so it's not such a risk, but I think the static method is more in keeping with our usual coding conventions around this kind of thing.

Reply:
Same for the `EntityDocument`.